bin/mail/2017-teacher-application-responses/teacher_application_decision_processor.rb

```diff
@@ -151,80 +151,57 @@ def load_workshop_map
 
   def process(decision, teacher_application, params = {})
     raise "Unexpected decision: #{decision}" unless @results.key? decision
+    params[:decision] = decision
 
-    # Make sure regional partner names have overrides applied before sending
-    if params.key? :regional_partner_name_s
-      params[:regional_partner_name_s] = apply_partner_name_overrides(params[:regional_partner_name_s])
-    end
+    email_params = Pd::TeacherApplicationEmailParams.new(teacher_application, params)
+    raise "Error: #{email_params.errors.inspect}" unless email_params.valid?
 
-    # Construct result hash, add to appropriate list, and return the new result
-    {
-      name: teacher_application.teacher_name,
-      email: teacher_application.primary_email,
-      preferred_first_name_s: teacher_application.teacher_first_name,
-      course_name_s: teacher_application.program_name
-    }.merge(params).tap do |result|
-      @results[decision] << result
+    email_params.to_final_params.except(:decision).tap do |effective_params|
+      @results[decision] << effective_params
     end
   end
 
-  def teachercon?(workshop_string)
-    TEACHER_CONS.any? {|tc| workshop_string.include? tc}
-  end
-
   def process_accept(teacher_application, program, accepted_workshop, regional_partner_override)
+    # First, update the actual dashboard DB with this accepted workshop string.
+    save_accepted_workshop teacher_application, program, accepted_workshop, regional_partner_override
+
     # There are 2 kinds of acceptance, TeacherCon (ours) and Regional Partner.
-    # We can tell based on the accepted workshop
-    if teachercon? accepted_workshop
-      process_accept_teachercon teacher_application, program, accepted_workshop, regional_partner_override
+    if teacher_application.accepted_program.teachercon?
```

Comment:
I prefer it this way since it's not returning anything, and to keep the line from being too long.